components/addTask.vue

```diff
@@ -47,6 +48,11 @@ import { defineComponent } from '@nuxtjs/composition-api'
 
 export default defineComponent({
   emits: ['newTask'],
+  data() {
+    return {
+      newTask: '',
+    }
+  },
   methods: {
     addTask() {
```

Comment:
This should have been an `async` function since you are dealing with promises here.